set.mm

```diff
@@ -25515,21 +25553,88 @@ replaced with setvar variables (see ~ cleljust ), so we don't include
+  Class membership
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    $d x y z t u v A $.  $d x y z t u v B $.
+    df-clel.1 $e |- ( y e. z <-> E. u ( u = y /\ u e. z ) ) $.
+    df-clel.2 $e |- ( t e. t <-> E. v ( v = t /\ v e. t ) ) $.
+    $( Define the membership connective between classes.  Theorem 6.3 of
+       [Quine] p. 41, or Proposition 4.6 of [TakeutiZaring] p. 13, which we
+       adopt as a definition.  See these references for its metalogical
+       justification.
+
+       The hypotheses express that all instances of the conclusion where class
+       variables are replaced with setvar variables hold.  Therefore, this
+       definition merely extends to class variables something that is true for
+       setvar variables, hence is conservative.  This is only a proof sketch of
+       conservativity; for details see Appendix of [Levy] p. 357.  This is the
+       reason why we call this axiomatic statement a "definition", even though
+       it does not have the usual form of a definition.  If we required a
+       definition to have the usual form, we would call ~ df-clel an axiom.
+
+       See also comments under ~ df-clab , ~ df-cleq , and ~ abeq2 .
+
+       Alternate characterizations of ` A e. B ` when either ` A ` or ` B ` is
+       a set are given by ~ clel2 , ~ clel3 , and ~ clel4 .
+
+       This is called the "axiom of membership" by [Levy] p. 338, who treats
+       the theory of classes as an extralogical extension to our logic and set
+       theory axioms.
+
+       While the three class definitions ~ df-clab , ~ df-cleq , and ~ df-clel
+       are eliminable and conservative and thus meet the requirements for sound
+       definitions, they are technically axioms in that they do not satisfy the
+       requirements for the current definition checker.  The proofs of
+       conservativity require external justification that is beyond the scope
+       of the definition checker.
+
+       For a general discussion of the theory of classes, see
+       ~ mmset.html#class .  (Contributed by NM, 26-May-1993.)  (Revised by BJ,
+       27-Jun-2019.) $)
+    df-clel $a |- ( A e. B <-> E. x ( x = A /\ x e. B ) ) $.
+  $}
+
+  ${
+    $d x y z t u v A $.  $d x y z t u v B $.
+    $( Characterization of the elements of a class.  (Contributed by BJ,
+       27-Jun-2019.) $)
+    dfclel $p |- ( A e. B <-> E. x ( x = A /\ x e. B ) ) $=
+      ( vy vz vv vu vt cleljust df-clel ) ADEFGHBCDEGIHHFIJ $.
+    $( $j usage 'dfclel' avoids 'ax-9' 'ax-10' 'ax-11' 'ax-12' 'ax-13'
+       'ax-ext' 'df-clab' 'df-cleq'; $)
+  $}
+
   ${
     $d z x $.  $d z y $.  $d z A $.
     $( Weaker version of ~ eleq1 (but more general than ~ elequ1 ) not
-       depending on ~ ax-ext nor ~ df-cleq .  (Contributed by BJ,
-       24-Jun-2019.) $)
+       depending on ~ ax-ext nor ~ df-cleq .  Note that this proof does not
+       depend on ~ ax-8 (while ~ eleq2w depends on ~ ax-9 ), so this also gives
```

Comment:
"if dfclel didn't use ax-8" is probably needed before "this proof does not depend on ax-8"